sinks/influxdb/driver.go

```diff
@@ -37,72 +37,130 @@ type influxdbSink struct {
+	return true
+}
 
-	// Add the real metric value.
-	columns = append(columns, "value")
-	values = append(values, timeseries.Point.Value)
-	// Append labels.
-	if !self.avoidColumns {
-		for key, value := range timeseries.Point.Labels {
-			columns = append(columns, key)
-			values = append(values, value)
+// Returns true if this sink supports events
+func (sink *influxdbSink) SupportsEvents() bool {
+	return true
+}
+
+func (sink *influxdbSink) StoreTimeseries(timeseriesLists []*[]sink_api.Timeseries) error {
+	precisions := []sink_api.TimePrecision{sink_api.Second, sink_api.Millisecond, sink_api.Microsecond}
+	for _, precision := range precisions {
+		err := sink.storeTimeseriesWithPrecision(timeseriesLists, precision)
+		if err != nil {
+			return err
 		}
-	} else {
-		seriesName = strings.Replace(seriesName, "/", "_", -1)
-		seriesName = fmt.Sprintf("%s_%s", sink_api.LabelsToString(timeseries.Point.Labels, "_"), seriesName)
 	}
-	// Add timestamp.
-	columns = append(columns, "time")
-	values = append(values, timeseries.Point.End.Unix())
-	// Ass sequence number
-	columns = append(columns, "sequence_number")
-	values = append(values, self.seqNum.Get(seriesName))
-
-	return self.newSeries(seriesName, columns, values)
+	return nil
 }
 
-func (self *influxdbSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+func (sink *influxdbSink) storeTimeseriesWithPrecision(timeseriesLists []*[]sink_api.Timeseries, precision sink_api.TimePrecision) error {
 	dataPoints := []*influxdb.Series{}
-	for index := range timeseries {
-		dataPoints = append(dataPoints, self.metricToSeries(&timeseries[index]))
+	for _, timeseriesList := range timeseriesLists {
+		if timeseriesList != nil {
+			for _, timeseries := range *timeseriesList {
+				if timeseries.TimePrecision == precision {
+					if !sink.avoidColumns {
+						dataPoints = append(dataPoints, sink.timeSeriesToInfluxdbSeries(&timeseries))
+					} else {
+						for _, point := range timeseries.Points {
+							dataPoints = append(dataPoints, sink.pointToInfluxdbSeriesNoColumn(timeseries.SeriesName, point))
+						}
+					}
+				}
+			}
+		}
 	}
+
 	// TODO: Group all datapoints belonging to a metric into a single series.
 	// TODO: Record the average time taken to flush data.
-	err := self.client.WriteSeriesWithTimePrecision(dataPoints, influxdb.Second)
-	if err != nil {
-		glog.Errorf("failed to write stats to influxDB - %s", err)
-		self.recordWriteFailure()
+	if len(dataPoints) > 0 {
+		err := sink.client.WriteSeriesWithTimePrecision(dataPoints, getInfluxdbPrecision(precision))
+		if err != nil {
+			glog.Errorf("failed to write time series to influxDB - %s", err)
+			for _, dataPoint := range dataPoints {
+				glog.Errorf("  detail: failed to write time series (%s) to influxDB with columns (%s)", dataPoint.Name, dataPoint.Columns)
+			}
+			sink.recordWriteFailure()
+			return err
+		}
+		glog.V(1).Info("Succesfully wrote %q timeseries to influxDB", precision)
```

Comment:
Hmm. Maybe rephrase to `Successfully wrote timeseries to InfluxDB with precision: %q`?